test/boost/reader_concurrency_semaphore_test.cc

```diff
@@ -1077,3 +1084,352 @@ SEASTAR_THREAD_TEST_CASE(test_reader_concurrency_semaphore_set_resources) {
+}
+
+// Check that the memory consumption limiting mechanism of the semaphore does prevent OOMs.
+// This test should be run with <1G memory.
+SEASTAR_TEST_CASE(test_reader_concurrency_semaphore_memory_limit_no_oom) {
+    auto db_cfg_ptr = make_shared<db::config>();
+    auto& db_cfg = *db_cfg_ptr;
+
+    db_cfg.enable_cache(false);
+    db_cfg.enable_commitlog(false);
+    db_cfg.reader_concurrency_semaphore_serialize_limit_multiplier.set(2);
+    db_cfg.reader_concurrency_semaphore_kill_limit_multiplier.set(4);
+
+    return do_with_cql_env_thread([] (cql_test_env& env) {
+        auto& db = env.local_db();
+
+        sstring value(256 * 1024, '0');
+
+        env.execute_cql("CREATE TABLE ks.tbl (pk int, ck int, value text, primary key (pk, ck)) WITH compaction = {'class': 'NullCompactionStrategy'};").get();
+
+        env.require_table_exists("ks", "tbl").get();
+
+        auto& tbl = db.find_column_family("ks", "tbl");
+        auto s = tbl.schema();
+        auto& sst_man = tbl.get_sstables_manager();
+        auto& semaphore = db.get_reader_concurrency_semaphore();
+
+        dht::decorated_key dk = [&] () -> dht::decorated_key {
+            std::optional<dht::decorated_key> dk;
+            int32_t i = 0;
+            do {
+                dk = dht::decorate_key(*s, partition_key::from_single_value(*s, serialized(i++)));
+            } while (dht::shard_of(*s, dk->token()) != this_shard_id());
+            return std::move(*dk);
+        }();
+
+        mutation mut(s, dk);
+        mut.set_clustered_cell(clustering_key::from_single_value(*s, serialized(0)), to_bytes("value"), data_value(value), 0);
+
+        auto sstables_dir = tmpdir();
+
+        const auto target_num_sstables = 32;
+        const auto sstable_write_concurrency = 16;
+        const auto num_reads = 64;
+
+        auto num_sstables = 0;
+        parallel_for_each(boost::irange(0, sstable_write_concurrency), [&] (int i) {
+            return seastar::async([&, i] {
+                while (num_sstables != target_num_sstables) {
+                    ++num_sstables;
+                    auto sst = sst_man.make_sstable(s, sstables_dir.path().string(), sstables::generation_type{num_sstables}, sst_man.get_highest_supported_format(), sstables::sstable_format_types::big);
+                    auto writer_cfg = sst_man.configure_writer("test");
+                    sst->write_components(
+                        make_flat_mutation_reader_from_mutations_v2(s, semaphore.make_tracking_only_permit(s.get(), "test", db::no_timeout), mut, s->full_slice()),
+                        1,
+                        s,
+                        writer_cfg,
+                        encoding_stats{}).get();
+                    sst->load().get();
+                    tbl.add_sstable_and_update_cache(std::move(sst)).get();
+                }
+            });
+        }).get();
+
+        auto read_id = env.prepare("SELECT value FROM ks.tbl WHERE pk = ? AND ck = ?").get0();
```

Comment:
Without BYPASS CACHE, why do we have any failures at all?

Maybe the reads all get admitted before the cache has a chance to cache anything. But better to add BYPASS CACHE to reduce the chance that it starts working based on load.

Reply:
I disabled cache globally (towards the start of the test):
```
    db_cfg.enable_cache(false);
```

Reply:
Ah, thanks. A comment explaining why would be good.

Reply:
(also drawing attention that this isn't just some cut'n'paste boilerplate)